openmdao/solvers/newton.py

```diff
@@ -116,6 +121,7 @@ def solve(self, params, unknowns, resids, system, metadata=None):
         system.children_solve_nonlinear(local_meta)
         system.apply_nonlinear(params, unknowns, resids)
 
+        unknowns_cache = np.zeros(unknowns.vec.shape)
```

Comment:
yeah, by the time that the solve setup() methods are called, all vectors have been allocated.